### PR TITLE
[Consul] Add health label to metrics

### PR DESCRIPTION
**Pull request by ShimmerGlass**
Label metrics with the target health using consul's /health endpoint.

### CODE REVIEW

File: discovery/consul/consul.go [hunk 1]
```diff
@@ -100,6 +102,10 @@ type SDConfig struct {
 	Username     string             `yaml:"username,omitempty"`
 	Password     config_util.Secret `yaml:"password,omitempty"`
 
+	// Only scrap services nodes this healthy, inclusive. Possible values are "critical", "warning",
+	// "passning".
+	// Defaults to "citical" (all services nodes) if empty.
+	MinHealth string `yaml:"min_health,omitempty"`
```
Comment (brian-brazil) [inline discovery/consul/consul.go]: This can be done with relabelling, and it is recommended to scrape nodes that are apparently unhealthy.

**Comment by Aestek** [comment]
Removed the `min_health` parameter. This change now only labels the metrics with the target health.

File: discovery/consul/consul.go [hunk 2]
```diff
@@ -481,48 +484,48 @@ func (srv *consulService) watch(ctx context.Context, ch chan<- []*targetgroup.Gr
...
 		// Add all key/value pairs from the service's metadata as their own labels.
-		for k, v := range node.ServiceMeta {
+		for k, v := range serviceNode.Service.Meta {
 			name := strutil.SanitizeLabelName(k)
 			labels[serviceMetaDataLabel+model.LabelName(name)] = model.LabelValue(v)
 		}
 
 		// Add all key/value pairs from the service's tagged addresses as their own labels.
-		for k, v := range node.TaggedAddresses {
+		for k, v := range serviceNode.Node.TaggedAddresses {
```
Comment (brian-brazil) [inline discovery/consul/consul.go]: This doesn't seem to exist for the Health call

Comment (Aestek) [reply discovery/consul/consul.go]: Hmm, it does : https://www.consul.io/api/health.html#sample-response-2

Comment (brian-brazil) [reply discovery/consul/consul.go]: Ah, I was looking in the Service rather than the Node

**Comment by ShimmerGlass** [comment]
Any update on this ?

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**